cardano_node_tests/utils/dbsync_utils.py

```diff
@@ -848,6 +849,168 @@ def check_param_proposal(protocol_params: dict) -> tp.Optional[dbsync_queries.Pa
+        "dvt_p_p_network_group": _evaluate_pparam(dvt.get("ppNetworkGroup")),
+        "dvt_p_p_technical_group": _evaluate_pparam(dvt.get("ppTechnicalGroup")),
+        "dvt_treasury_withdrawal": _evaluate_pparam(dvt.get("treasuryWithdrawal")),
+        "dvt_update_to_constitution": _evaluate_pparam(dvt.get("updateToConstitution")),
+        # - Pools
+        "pvt_committee_no_confidence": _evaluate_pparam(pvt.get("committeeNoConfidence")),
+        "pvt_committee_normal": _evaluate_pparam(pvt.get("committeeNormal")),
+        "pvt_hard_fork_initiation": _evaluate_pparam(pvt.get("hardForkInitiation")),
+        "pvt_motion_no_confidence": _evaluate_pparam(pvt.get("motionNoConfidence")),
+        "pvtpp_security_group": _evaluate_pparam(pvt.get("ppSecurityGroup")),
+        # General
+        "gov_action_lifetime": pp.get("govActionLifetime"),
+        "gov_action_deposit": pp.get("govActionDeposit"),
+        "drep_deposit": pp.get("dRepDeposit"),
+        "drep_activity": pp.get("dRepActivity"),
+        "committee_min_size": pp.get("committeeMinSize"),
+        "committee_max_term_length": pp.get("committeeMaxTermLength"),
+    }
+
+    return params_mapping
+
+
+def _check_param_proposal(
+    param_proposal_db: tp.Union[dbsync_queries.ParamProposalDBRow, dbsync_queries.EpochParamDBRow],
+    params_map: dict,
+) -> list:
+    """Check parameter proposal against db-sync."""
+    failures = []
+
+    for param_db, protocol_value in params_map.items():
+        if protocol_value:
+            db_value = getattr(param_proposal_db, param_db)
+            if db_value and (db_value != protocol_value):
+                failures.append(
+                    f"Param value for {param_db}: {db_value}. Expected: {protocol_value}"
+                )
+    return failures
+
+
+def check_conway_param_update_proposal(
+    param_proposal_ledger: dict,
+) -> tp.Optional[dbsync_queries.ParamProposalDBRow]:
+    """Check comparison for param proposal between ledger and db-sync."""
+    param_proposal_db = dbsync_queries.query_param_proposal()
+    params_map = map_params_to_db_convention(param_proposal_ledger)
+    failures = []
+
+    # Get cost models
+    if param_proposal_db.cost_model_id:
+        db_cost_model = dbsync_queries.query_cost_model(param_proposal_db.cost_model_id)
+        pp_cost_model = param_proposal_ledger.get("costModels")
+        if db_cost_model != pp_cost_model:
+            failures.append(f"Cost model mismatch for {db_cost_model}. Expected: {pp_cost_model}")
+    failures.extend(_check_param_proposal(param_proposal_db, params_map))
+
+    if failures:
+        failures_str = "\n".join(failures)
+        msg = f"Unexpected parameter proposal values in db-sync:\n{failures_str}"
+        raise AssertionError(msg)
+    return param_proposal_db
+
+
+def check_conway_param_update_enactment(
+    gov_state: dict, epoch_no: int
+) -> tp.Optional[dbsync_queries.EpochParamDBRow]:
```

Comment:
For functions called directly from tests, we need to check that we have db-sync available.